toptal/picasso
# Fix watch build scripts

### Description

We are adding a new script `build:package:watch` which runs the lerna script `build:package` for packages, instead of solely `tsc` (some packages need extra steps when running build). I also added local caching for the `build:package` script by default, so we can have quicker builds on local development.

Scripts that had to watch build, like storybook and integration tests now reference `build:package:watch` instead and have a better view of the commands being run.

The `build:all` script was also renamed to `tsc:all`, to make the intent of it more clear

### How to test


- [Temploy](https://picasso.toptal.net/feature/better-scripts)
- CI should be green
- Checkout the code and try running `yarn build:package` more than once, it should cache it the first time
- Running `yarn build:package:watch` should rebuild packages on change
- Running `yarn start` will build every package and start watching for changes with concurrently

### Screenshots

![image](https://github.com/toptal/picasso/assets/5845160/d9a4c21c-9ebf-46eb-9a17-7eb61ad578e5)
Cached builds

### Development checks


**Breaking change**


> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>
